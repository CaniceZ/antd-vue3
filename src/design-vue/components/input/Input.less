.ygp-input {
  & .ant-input[disabled] {
    color: @text-color-disabled;
  }
  &.ant-input[disabled] {
    color: @text-color-disabled;
  }
}
