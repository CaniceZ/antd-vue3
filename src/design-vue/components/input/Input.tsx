import {
  InputProps as AInputProps,
  InputNumberProps as AInputNumberProps,
  TextAreaProps as ATextAreaProps
} from 'ant-design-vue'
import { getPrefix, isAttrTrue } from '../../_utils/common'
import {
  defineComponent,
  PropType,
  h,
  resolveComponent,
  onMounted,
  nextTick
} from 'vue'
import useRefExpose from '../../_hooks/useRefExpose'
import {
  AInputExpose,
  AInputPasswordProps,
  AInputSearchProps,
  AInputSlots
} from './type'
import { GlobalComponentConstructor, PropsType } from '../../types/ts-helpers'

type InputType =
  | 'text'
  | 'input'
  | 'number'
  | 'password'
  | 'search'
  | 'textarea'

const [prefixName] = getPrefix('input')

const Input = defineComponent({
  name: prefixName,
  inheritAttrs: false,
  props: {
    modelValue: {
      type: [String, Number]
    },
    type: {
      type: String as PropType<InputType>,
      default: ''
    },
    number: {
      type: Boolean,
      default: false
    },
    password: {
      type: Boolean,
      default: false
    },
    search: {
      type: Boolean,
      default: false
    },
    textarea: {
      type: Boolean,
      default: false
    },
    /* 是否自动填充精确度的小数0 */
    fillPrecision: {
      type: Boolean,
      default: false
    }
  },
  emits: ['update:modelValue'],
  setup(props, { attrs, slots, expose, emit }) {
    const [inputRef, inputExpose] = useRefExpose<AInputExpose>([
      'blur',
      'focus',
      'setValue'
    ])

    expose(inputExpose)

    onMounted(() => {
      nextTick(() => {
        // TODO fix 需要排除 disabled 和 readonly 的情况
        if (attrs['autofocus']) {
          inputRef.value?.focus()
        }
      })
    })

    function getComponent(t?: InputType) {
      if (t === 'search' && isAttrTrue(attrs.isAutoComplete)) {
        return getComponent()
      }
      let type: InputType
      if (t) {
        type = t
      } else if (props.number) {
        type = 'number'
      } else if (props.password) {
        type = 'password'
      } else if (props.search) {
        type = 'search'
      } else if (props.textarea) {
        type = 'textarea'
      } else {
        type = 'input'
      }
      if (['text', 'input'].includes(type)) {
        return 'input'
      } else if (['number', 'password', 'search'].includes(type)) {
        return `input-${type}`
      } else {
        return type
      }
    }

    function formatter(
      value: number | string,
      { userTyping, input }: { userTyping: boolean; input: string }
    ): string {
      if (userTyping) {
        return input
      }
      return typeof value === 'number' ? String(value) : value
    }

    return () => {
      const type = getComponent(props.type)
      const Component = resolveComponent(`a-${type}`) as string
      return h(
        Component,
        {
          htmlSize: 20,
          formatter:
            type === 'input-number' && !props.fillPrecision
              ? formatter
              : void 0,
          ...attrs,
          value: props.modelValue,
          'onUpdate:value': val => emit('update:modelValue', val),
          class: [`ygp-${type}`, attrs.class],
          ref: inputRef
        },
        slots
      )
    }
  }
})

export type InputProps = PropsType<typeof Input> &
  (
    | PropsType<AInputProps>
    | PropsType<ATextAreaProps>
    | PropsType<AInputNumberProps>
    | PropsType<AInputSearchProps>
    | PropsType<AInputPasswordProps>
  )
export type InputExpose = AInputExpose
export type InputSlots = AInputSlots

export default Input as GlobalComponentConstructor<InputProps, InputSlots>
