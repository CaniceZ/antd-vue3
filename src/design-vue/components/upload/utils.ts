export const extname = (url = '') => {
  const temp = url.split('/')
  const filename = temp[temp.length - 1]
  const filenameWithoutSuffix = filename.split(/#|\?/)[0]
  return (/\.[^./\\]*$/.exec(filenameWithoutSuffix) || [''])[0]
}

const isImageFileType = type => !!type && type.indexOf('image/') === 0

export const isImageUrl = file => {
  if (isImageFileType(file.type)) {
    return true
  }
  const url = file.thumbUrl || file.url
  const extension = extname(url)
  if (
    /^data:image\//.test(url) ||
    /(webp|svg|png|gif|jpg|jpeg|jfif|bmp|dpg|ico)$/i.test(extension)
  ) {
    return true
  }
  if (/^data:/.test(url)) {
    // other file types of base64
    return false
  }
  if (extension) {
    // other file types which have extension
    return false
  }
  return true
}

export const getFileType = file => {
  if (!file) {
    return
  }
  const attachmentUrl = file.url

  const type = extname(attachmentUrl)
  let fileType = ''
  if (
    isImageFileType(file.type) ||
    /(webp|svg|png|gif|jpg|jpeg|jfif|bmp|dpg|ico)$/i.test(type)
  ) {
    fileType = 'image'
  } else if (type === '.pdf') {
    fileType = 'pdf'
  } else if (['.doc', '.dot'].includes(type)) {
    fileType = 'word'
  } else if (['.xls', '.xlsx'].includes(type)) {
    fileType = 'excel'
  } else {
    fileType = 'file'
  }
  return fileType
}

export const imageDimensions = file =>
  new Promise((resolve, reject) => {
    const img = new Image()

    img.onload = () => {
      const { naturalWidth: width, naturalHeight: height } = img
      resolve({ width, height })
    }

    img.onerror = () => {
      reject('There was some problem with the image.')
    }

    img.src = URL.createObjectURL(file)
  })

export const isUrl = (url: string) => {
  const regex = /^(http|https)/
  return regex.test(url.toLowerCase())
}
