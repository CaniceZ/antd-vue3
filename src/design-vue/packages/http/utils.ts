const toString = Object.prototype.toString

export function is(val: unknown, type: string) {
  return toString.call(val) === `[object ${type}]`
}

export function isString(val: unknown): val is string {
  return is(val, 'String')
}

export function isObject(val: any): val is Record<any, any> {
  return val !== null && is(val, 'Object')
}

export function isFunction(val: unknown): val is Function {
  return typeof val === 'function'
}

/**
 * Add the object as a parameter to the URL
 * @param baseUrl url
 * @param obj
 * @returns {string}
 * eg:
 *  let obj = {a: '3', b: '4'}
 *  setObjToUrlParams('www.baidu.com', obj)
 *  ==>www.baidu.com?a=3&b=4
 */
export function setObjToUrlParams(baseUrl: string, obj: any): string {
  let parameters = ''
  for (const key in obj) {
    parameters += key + '=' + encodeURIComponent(obj[key]) + '&'
  }
  parameters = parameters.replace(/&$/, '')
  return /\?$/.test(baseUrl)
    ? baseUrl + parameters
    : baseUrl.replace(/\/?$/, '?') + parameters
}

export function deepMerge<T = any>(src: any = {}, target: any = {}): T {
  let key: string
  for (key in target) {
    src[key] = isObject(src[key])
      ? deepMerge(src[key], target[key])
      : (src[key] = target[key])
  }
  return src
}

declare global {
  interface Navigator {
    msSaveBlob(blob: Blob, string): void
  }
}

/**
 * 根据后端接口文件流下载
 * @param {*} data
 * @param {*} filename
 * @param {*} mime
 * @param {*} bom
 */
export function downloadByData(
  data: BlobPart,
  filename: string,
  mime?: string,
  bom?: BlobPart
) {
  const blobData = typeof bom !== 'undefined' ? [bom, data] : [data]
  const blob = new Blob(blobData, { type: mime || 'application/octet-stream' })
  // 如果浏览器的msSaveBlob存在
  if (typeof window.navigator.msSaveBlob !== 'undefined') {
    // 使用msSaveBlob下载文件
    window.navigator.msSaveBlob(blob, filename)
  } else {
    // 创建一个URL
    const blobURL = window.URL.createObjectURL(blob)
    const tempLink = document.createElement('a')
    tempLink.style.display = 'none'
    tempLink.href = blobURL
    tempLink.setAttribute('download', filename)
    if (typeof tempLink.download === 'undefined') {
      tempLink.setAttribute('target', '_blank')
    }
    document.body.appendChild(tempLink)
    // 手动触发点击事件
    tempLink.click()
    // 点击完成后移除链接
    document.body.removeChild(tempLink)
    // 移出URL
    window.URL.revokeObjectURL(blobURL)
  }
}
