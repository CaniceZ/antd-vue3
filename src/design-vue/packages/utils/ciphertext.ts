import get from 'lodash-es/get'

type Options = {
  sign: '*' | string
  suffix: '_encrypt' | string
}

const defaultOptions: Options = {
  sign: '*',
  suffix: '_encrypt'
}

/**
 * 获取密文的密钥字段名
 * @param name
 * @param opts
 * @returns
 */
export function getCiphertextField(
  name: string,
  opts: Partial<Options> = defaultOptions
): string {
  const options = {
    ...defaultOptions,
    ...opts
  }
  return `${name}${options.suffix}`
}

/**
 * 获取密文对象实例
 * @param model 数据对象
 * @param name 密文字段名称 / 密文字段路径
 * @param opts.sign 脱敏显示符号
 * @param opts.suffix 脱敏密文字段名后缀
 * @returns
 */
export function getCiphertext(
  model: object,
  name: string,
  opts: Partial<Options> = defaultOptions
): {
  text: string
  cipherText: string
} {
  const options = {
    ...defaultOptions,
    ...opts
  }
  const emptyRes = {
    text: '',
    cipherText: ''
  }
  if (typeof model === 'object') {
    const text = get(model, name)
    const cipherText = get(model, getCiphertextField(name, options))
    if (text && cipherText && text.includes(options.sign)) {
      return {
        text,
        cipherText
      }
    } else {
      return emptyRes
    }
  } else {
    return emptyRes
  }
}

/**
 * 判断是否密文
 * @param model 数据对象
 * @param name 密文字段名称 / 密文字段路径
 * @param opts.sign 脱敏显示符号
 * @param opts.suffix 脱敏密文字段名后缀
 */
export function isCiphertext(
  model: object,
  name: string,
  opts: Partial<Options> = defaultOptions
): boolean {
  const options = {
    ...defaultOptions,
    ...opts
  }
  const ciphertext = getCiphertext(model, name, options)
  return Boolean(ciphertext.text && ciphertext.cipherText)
}
