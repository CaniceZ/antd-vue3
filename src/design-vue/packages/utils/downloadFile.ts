/**
 * 下载文件
 * @param url 下载路径
 * @returns void
 */
export default function downloadFile(url: string) {
  if (!url) return
  const a = document.createElement('a')
  a.href = url
  a.click()
}
