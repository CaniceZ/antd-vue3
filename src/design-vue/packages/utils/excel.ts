import { Message, Modal, YImportResultModal } from '../../'

/**
 * 导入文件
 * @param params
 * @param callback
 * @returns Promise
 */
export const importFile = async (params: {
  file: File | any
  title: string
  getBatchNumber: Function
  getImportResult: Function
  wrapCount?: number
  extraData?: Object
  templateCode?: string
  resultProps?: any
  resultSlots?: any
}) => {
  const {
    file,
    getBatchNumber,
    getImportResult,
    title,
    extraData,
    templateCode,
    resultProps,
    resultSlots,
    wrapCount
  } = params

  const formData = new FormData()
  formData.append('file', file)
  templateCode && formData.append('templateCode', templateCode) // 模板编码

  if (extraData) {
    Object.keys(extraData).forEach(key => {
      formData.append(key, extraData![key])
    })
  }

  const res: any = await getBatchNumber(formData)
  return new Promise((resolve, reject) => {
    const batchNumber = res?.batchNumber || res
    if (batchNumber) {
      Message.success('上传成功，稍后通知您导入结果!')
      importResultPolling({
        resultProps,
        resultSlots,
        getImportResult,
        batchNumber,
        title,
        wrapCount,
        doneCallback: () => {
          resolve(params)
        },
        timeoutCallback: () => {
          reject()
        }
      })
    } else {
      Message.error('上传失败，请稍后重试')
      reject()
    }
  })
}

/**
 * 导入结果轮询
 * @param params 轮询的请求参数
 * @param callback 成功回调
 */
export const importResultPolling = (params: {
  title: String
  batchNumber: string
  getImportResult: Function
  resultProps?: any
  resultSlots?: any
  wrapCount?: number
  doneCallback?: (...args: any) => void
  timeoutCallback?: (...args: any) => void
}) => {
  const {
    wrapCount,
    batchNumber,
    title,
    doneCallback,
    timeoutCallback,
    resultProps,
    resultSlots,
    getImportResult
  } = params
  if (!batchNumber) return
  function run(count = 1) {
    setTimeout(() => {
      let params = { batchNumber }
      if (resultProps?.extraData) {
        params = { ...params, ...resultProps.extraData }
      }
      const typeMap = resultProps?.typeMap
      getImportResult(params).then(
        res => {
          if (resultProps?.isPms) {
            // pms特殊导入结果处理
            if (res && res.id) {
              if (!resultProps?.hideModal) {
                Modal.create(YImportResultModal, {
                  list: [res],
                  typeMap,
                  importBizSet: [resultProps.importBiz],
                  title,
                  ...(resultProps || {})
                })
              }
              doneCallback?.({ res, typeMap, title })
            } else {
              if (count < (wrapCount || 20)) {
                run(++count)
              } else {
                Message.error('稍后请在最近导入结果中查看')
                timeoutCallback?.()
              }
            }
          } else {
            // excel通用
            const { status, msg } = res
            if (status === 4) {
              if (count < (wrapCount || 20)) {
                run(++count)
              } else {
                Message.error(msg || '稍后请在最近导入结果中查看')
                timeoutCallback?.()
              }
            } else {
              if (resultProps?.callback) {
                resultProps.callback(res)
              } else {
                if (!resultProps?.hideModal) {
                  Modal.create(
                    YImportResultModal,
                    {
                      title,
                      list: [res],
                      ...(resultProps || {})
                    },
                    resultSlots || {}
                  )
                }
                doneCallback?.({ res, typeMap, title })
              }
            }
          }
        },
        err => {
          Message.error(err?.msg || '上传失败，请稍后重试')
        }
      )
    }, 3000)
  }
  run()
}

// pms导出结果轮询
export const pmsExportPolling = (option: {
  batchNumber: string
  exportBiz: String
  userCode: String
  getPmsExportDetail: Function
  timeout?: (...args: any) => void
  done?: (...args: any) => void
  error?: (...args: any) => void
}) => {
  const {
    batchNumber,
    exportBiz,
    userCode,
    getPmsExportDetail,
    timeout,
    done,
    error
  } = option
  if (!batchNumber) return
  const params = {
    batchNumber: batchNumber,
    exportBiz,
    userCode
  }
  function run(count = 1) {
    setTimeout(() => {
      getPmsExportDetail(params).then(
        res => {
          const { code, data, msg } = res
          if (code === 0) {
            if (data && data.list.length > 0) {
              done?.(data)
              data.list &&
                data.list.forEach(item => {
                  if (item.excelUrl) {
                    window.location.href = item.excelUrl
                  } else {
                    Message.error(item.msg)
                  }
                })
            } else {
              if (count < 40) {
                run(++count)
              } else {
                Message.error('导出超时，请重新导出')
                timeout?.()
              }
            }
          } else {
            Message.error(msg)
            error?.()
          }
        },
        () => {
          error?.()
        }
      )
    }, 3000)
  }
  run()
}
