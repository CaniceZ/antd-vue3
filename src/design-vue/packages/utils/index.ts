import omit from './omit'
import pick from './pick'
import downloadFile from './downloadFile'
import { isCiphertext, getCiphertext, getCiphertextField } from './ciphertext'
import { needCLodop, loadCLodop, getLodop } from './LodopFuncs'
import { importFile, importResultPolling } from './excel'
import filterUnit from './filterUnit'
import filterMoney from './filterMoney'

export {
  omit,
  pick,
  downloadFile,
  isCiphertext,
  getCiphertext,
  getCiphertextField,
  needCLodop,
  loadCLodop,
  getLodop,
  importFile,
  importResultPolling,
  filterUnit,
  filterMoney
}
