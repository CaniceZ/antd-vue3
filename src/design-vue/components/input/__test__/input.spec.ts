import { mount } from '@vue/test-utils'
import { asyncExpect } from '../../../tests/utils'
import Input from '..'

// import Form from '../../form';
// import focusTest from '../../../tests/shared/focusTest'
// import { AInput } from '@ant-design';

// const { TextArea, Password } = Input

function commonType(type, output) {
  return it(`this input type is ${type}`, async () => {
    const wrapper = mount(Input as any, {
      props: { type }
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.classes()).toEqual(expect.arrayContaining([output]))
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })
}

describe('YInput', () => {
  commonType('password', 'ygp-input-password')
  commonType('search', 'ygp-input-search')
  commonType('number', 'ygp-input-number')
  commonType('text', 'ygp-input')
  commonType('input', 'ygp-input')
  // commonType('textarea', 'ygp-textarea')
  it(`this input type is textarea`, async () => {
    const wrapper = mount(Input as any, {
      props: { textarea: true, autoSize: { minRows: 2, maxRows: 6 }, rows: 1 }
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.classes()).toEqual(expect.arrayContaining(['ygp-textarea']))
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })

  it('should support fillPrecision', async () => {
    const wrapper = mount(Input as any, {
      props: {
        number: true,
        precision: 6,
        fillPrecision: true
      },
      sync: false
    })
    const passwordInput: any = wrapper.find('.ant-input-number-input')
    await wrapper.vm.$nextTick()
    // await wrapper.setData({ modelValue: 1.0 })
    await passwordInput.setValue(3)
    await passwordInput.trigger('blur')
    // expect(wrapper.vm.$props.number).toBeTypeOf('boolean')
    expect(passwordInput.element.value).toBe('3.000000')
    // expect(wrapper.vm.$props.number).toBeTypeOf('boolean')
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })
})
