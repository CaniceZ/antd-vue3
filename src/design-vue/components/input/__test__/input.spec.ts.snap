// Vitest Snapshot v1

exports[`YInput > should support fillPrecision 1`] = `
"<div class=\\"ant-input-number ygp-input-number\\">
  <div class=\\"ant-input-number-handler-wrap\\"><span unselectable=\\"on\\" role=\\"button\\" aria-label=\\"Increase Value\\" aria-disabled=\\"false\\" class=\\"ant-input-number-handler ant-input-number-handler-up\\"><span role=\\"img\\" aria-label=\\"up\\" class=\\"anticon anticon-up ant-input-number-handler-up-inner\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z\\"></path></svg></span></span><span unselectable=\\"on\\" role=\\"button\\" aria-label=\\"Decrease Value\\" aria-disabled=\\"false\\" class=\\"ant-input-number-handler ant-input-number-handler-down\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down ant-input-number-handler-down-inner\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div>
  <div class=\\"ant-input-number-input-wrap\\"><input autocomplete=\\"off\\" role=\\"spinbutton\\" step=\\"1\\" htmlsize=\\"20\\" class=\\"ant-input-number-input\\" aria-valuenow=\\"3\\"></div>
</div>"
`;

exports[`YInput > this input type is input 1`] = `"<input type=\\"text\\" class=\\"ant-input ygp-input\\">"`;

exports[`YInput > this input type is number 1`] = `
"<div class=\\"ant-input-number ygp-input-number\\">
  <div class=\\"ant-input-number-handler-wrap\\"><span unselectable=\\"on\\" role=\\"button\\" aria-label=\\"Increase Value\\" aria-disabled=\\"false\\" class=\\"ant-input-number-handler ant-input-number-handler-up\\"><span role=\\"img\\" aria-label=\\"up\\" class=\\"anticon anticon-up ant-input-number-handler-up-inner\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z\\"></path></svg></span></span><span unselectable=\\"on\\" role=\\"button\\" aria-label=\\"Decrease Value\\" aria-disabled=\\"false\\" class=\\"ant-input-number-handler ant-input-number-handler-down\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down ant-input-number-handler-down-inner\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div>
  <div class=\\"ant-input-number-input-wrap\\"><input autocomplete=\\"off\\" role=\\"spinbutton\\" step=\\"1\\" htmlsize=\\"20\\" class=\\"ant-input-number-input\\"></div>
</div>"
`;

exports[`YInput > this input type is password 1`] = `"<span class=\\"ant-input-affix-wrapper ant-input-password ygp-input-password\\"><!----><input type=\\"password\\" class=\\"ant-input\\"><span class=\\"ant-input-suffix\\"><!----><!----><span role=\\"img\\" aria-label=\\"eye-invisible\\" tabindex=\\"-1\\" class=\\"anticon anticon-eye-invisible ant-input-password-icon\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"eye-invisible\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z\\"></path><path d=\\"M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z\\"></path></svg></span></span></span>"`;

exports[`YInput > this input type is search 1`] = `"<span class=\\"ant-input-group-wrapper ant-input-search ygp-input-search\\"><span class=\\"ant-input-wrapper ant-input-group\\"><!----><input type=\\"text\\" class=\\"ant-input\\"><span class=\\"ant-input-group-addon\\"><button class=\\"ant-btn ant-input-search-button ant-btn-icon-only\\" type=\\"button\\"><span role=\\"img\\" aria-label=\\"search\\" class=\\"anticon anticon-search\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"search\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></span></button></span></span></span>"`;

exports[`YInput > this input type is text 1`] = `"<input type=\\"text\\" class=\\"ant-input ygp-input\\">"`;

exports[`YInput > this input type is textarea 1`] = `"<textarea htmlsize=\\"20\\" rows=\\"1\\" class=\\"ant-input ygp-textarea\\" style=\\"height: -24px; min-height: -8px; max-height: -24px; overflow-y: hidden; resize: none; overflow-x: hidden;\\"></textarea>"`;
